packages/preview-middleware/src/base/flp.ts

```diff
@@ -196,20 +196,23 @@ export class FlpSandbox {
             },
             locateReuseLibsScript: this.config.libs ?? (await this.hasLocateReuseLibsScript())
         };
+        const id = manifest['sap.app'].id;
         this.addApp(manifest, {
             componentId,
-            target: resources[componentId ?? manifest['sap.app'].id] ?? this.templateConfig.basePath,
+            target: resources[componentId ?? id] ?? this.templateConfig.basePath,
             local: '.',
             intent: this.config.intent
         });
         this.addStandardRoutes();
         if (this.rta) {
+            const appTitle = manifest['sap.app'].title;
             this.rta.options ??= {};
-            this.rta.options.baseId = componentId ?? manifest['sap.app'].id;
+            this.rta.options.baseId = componentId ?? id;
+            this.rta.options.appName = appTitle === '{{appTitle}}' ? 'My Application' : appTitle || id;
```

Comment:
I am not sure if this is too much overhead just to show a meaningful title. I would propose to just use the app ID. Let's discuss it today in our sync.

Reply:
As agreed we will just use the app ID